migrations/sql/V2019.06.26.10.30__add_code_required_report_tables.sql

```diff
@@ -0,0 +1,154 @@
+    description                          varchar(300)                            NOT NULL   ,
+    compliance_article_id                integer                                 NOT NULL   ,
+    due_date_period_months               integer                                 NOT NULL   ,
+    mine_report_due_date_type            character varying(3)                    NOT NULL   ,
+    active_ind                           boolean DEFAULT true                    NOT NULL   ,
+    create_user                          character varying(60)                   NOT NULL   ,
+    create_timestamp                     timestamp with time zone DEFAULT now()  NOT NULL   ,
+    update_user                          character varying(60)                   NOT NULL   ,
+    update_timestamp                     timestamp with time zone DEFAULT now()  NOT NULL   ,
+
+    FOREIGN KEY (compliance_article_id) REFERENCES compliance_article(compliance_article_id) DEFERRABLE INITIALLY DEFERRED,
+    FOREIGN KEY (mine_report_due_date_type) REFERENCES mine_report_due_date_type(mine_report_due_date_type) DEFERRABLE INITIALLY DEFERRED
+);
+
+COMMENT ON TABLE mine_report_definition is 'Definitions of code required reports, due dates and articles of the code that apply to them.';
+ALTER TABLE mine_report_definition OWNER TO mds;
+
+CREATE TABLE IF NOT EXISTS mine_report
+(
+    mine_report_id               serial                                   NOT NULL PRIMARY KEY,
+    mine_report_guid             uuid                                     NOT NULL            ,
+    mine_report_definition_id    integer                                  NOT NULL            ,
+    mine_guid                    uuid                                     NOT NULL            ,
+    permit_id                    integer                                                      ,
+    due_date                     timestamp                                NOT NULL            ,
+    submission_year              integer                                                      ,
+    deleted_ind                  boolean DEFAULT false                    NOT NULL            ,
+    create_user                  character varying(60)                    NOT NULL            ,
+    create_timestamp             timestamp with time zone DEFAULT now()   NOT NULL            ,
+    update_user                  character varying(60)                    NOT NULL            ,
+    update_timestamp             timestamp with time zone DEFAULT now()   NOT NULL            ,
+    
+    FOREIGN KEY (mine_report_definition_id) REFERENCES mine_report_definition(mine_report_definition_id) DEFERRABLE INITIALLY DEFERRED,
+    FOREIGN KEY (mine_guid) REFERENCES mine(mine_guid) DEFERRABLE INITIALLY DEFERRED,
+    FOREIGN KEY (permit_id) REFERENCES permit(permit_id) DEFERRABLE INITIALLY DEFERRED
+
+);
+
+COMMENT ON TABLE mine_report is 'A code required report for a specific mine or permit on a mine.';
```

Comment:
I'm wondering if we should be more specific about what this table is... I am still a little foggy on it... but is it essentially a list of mandatory reports that must be submitted by a mine as required by the Health, Safety and Reclamation Code? We've been using the term "code required reports" and I'm thinking we should define, somewhere, what this means is a more non-business-jargon way...

Reply:
I tried to make the comment more specific or more understandable.